src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5429,9 +5438,27 @@ public override BoundNode VisitDiscardExpression(BoundDiscardExpression node)
 
         public override BoundNode VisitThrowExpression(BoundThrowExpression node)
         {
-            var result = base.VisitThrowExpression(node);
-            SetUnknownResultNullability();
-            return result;
+            VisitThrow(node.Expression);
+            return null;
+        }
+
+        public override BoundNode VisitThrowStatement(BoundThrowStatement node)
+        {
+            VisitThrow(node.ExpressionOpt);
+            return null;
+        }
+
+        private void VisitThrow(BoundExpression expr)
+        {
+            var result = VisitRvalueWithResult(expr);
+            if ((expr.ConstantValue?.IsNull == true && !expr.IsSuppressed) ||
```

Comment:
Can `expr` be `null` here? The input to this method in `VisitThrowStatement` is `ExpressionOpt` which generally means `null` is a possible value.

Reply:
Indeed. Let me try that.